<documentation title="Removed PHP 4 Style Constructors">
    <standard><![CDATA[
Declaration of a PHP 4 style class constructor is deprecated since PHP 7.0 and removed since PHP 8.0.
]]></standard>
    <code_comparison>
        <code title="Valid: new style"><![CDATA[
class A
{
    function __construct()
    {
    }
}
        ]]></code>
        <code title="Invalid: old style"><![CDATA[
class A
{
    function A()
    {
    }
}
        ]]></code>
    </code_comparison>
    <code_comparison>
        <code title="Valid: calling direct parent"><![CDATA[
class A extends B
{
    function __construct()
    {
        parent::__construct();
    }
}

class B
{
    function __construct() {}
}
        ]]></code>
        <code title="Invalid: calling direct parent"><![CDATA[
class A extends B
{
    function A()
    {
        $this->B();
    }
}

class B
{
    function B() {}
}
        ]]></code>
    </code_comparison>
    <code_comparison>
        <code title="Valid: calling indirect parent"><![CDATA[
class A extends B
{
    function __construct()
    {
        C::__construct();
    }
}

class B extends C
{
    function __construct()
    {
        throw new Exception('Do not call');
    }
}

class C
{
    function __construct() {}
}
        ]]></code>
        <code title="Invalid: calling indirect parent"><![CDATA[
class A extends B
{
    function A()
    {
        $this->C();
    }
}

class B extends C
{
    function B()
    {
        throw new Exception('Do not call');
    }
}

class C
{
    function C() {}
}
        ]]></code>
    </code_comparison>
    <link>https://www.zend.com/php-migration/function-name-restrictions/removed-php4-style-constructor</link>
</documentation>
